src/dos/program_mount.cpp

```diff
@@ -350,71 +359,83 @@ void MOUNT::Run(void) {
 			if (temp_line == "/") WriteOut(MSG_Get("PROGRAM_MOUNT_WARNING_OTHER"));
 #endif
 			if (type == "overlay") {
-				localDrive *ldp = dynamic_cast<localDrive *>(
-						Drives[drive_index(drive)]);
-				cdromDrive *cdp = dynamic_cast<cdromDrive *>(
-						Drives[drive_index(drive)]);
+				const auto ldp = dynamic_cast<localDrive*>(Drives[drive_index(drive)]);
+				const auto cdp = dynamic_cast<cdromDrive*>(Drives[drive_index(drive)]);
 				if (!ldp || cdp) {
 					WriteOut(MSG_Get("PROGRAM_MOUNT_OVERLAY_INCOMPAT_BASE"));
 					return;
 				}
 				std::string base = ldp->GetBasedir();
 				uint8_t o_error = 0;
-				newdrive = new Overlay_Drive(base.c_str(),temp_line.c_str(),sizes[0],int8_tize,sizes[2],sizes[3],mediaid,o_error);
-				//Erase old drive on success
+				newdrive = std::make_unique<Overlay_Drive>(
+				        base.c_str(),
+				        temp_line.c_str(),
+				        sizes[0],
+				        int8_tize,
+				        sizes[2],
+				        sizes[3],
+				        mediaid,
+				        o_error);
```

Comment:
These function calls give me the creeps... What is all this rubbish? They should've just passed in a struct instead of EIGHT function parameters... 😢

Reply:
yes :-) `OverLay_Drive` class constructor.. inherited from `localDrive` class constructor .. inherited from `DOS_Drive` constructor..

Reply:
Man... 😐